### PR TITLE
Conan v2 support for nanodbc

### PR DESCRIPTION
**Pull request by zillemarco**
Specify library name and version:  **nanodbc** (all versions)

nanodbc was present but still not working for Conan v2.

This PR changes the package so it can be used with Conan v2.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by zillemarco** [comment]
Hi @prince-chrismc 👋 🙂 This is now ready for a review 🙂 Could you have a look and add it to the list open PRs to review? 🙏 Thanks a lot 🙂

**Review by SSE4 — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]
LGTM, thanks for your contribution!
I've restored the logic to check for minimum C++ standard to keep the recipe compatible with Conan 1.x.

**Review by davidsanfal — DISMISSED** [review]

File: recipes/nanodbc/all/conanfile.py [hunk 1]
```diff
@@ -12,7 +14,6 @@ class NanodbcConan(ConanFile):
     homepage = "https://github.com/nanodbc/nanodbc/"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
```
Comment (franramirez688) [inline recipes/nanodbc/all/conanfile.py:16]: I think that we could add the `package_type` here:

```python
package_type = "library"
```

Comment (jcar87) [reply recipes/nanodbc/all/conanfile.py:17]: Very right! Just added this

**Review by franramirez688 — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Pull request merged**